components/Participants/ParticipantSelector.jsx

```diff
@@ -0,0 +1,85 @@
+import Checkbox from "./Checkbox";
+import { ScrollView, View, Text } from "react-native";
+import ProfilePicture from "../ProfileElements/ProfilePicture";
+import { useEffect, useState } from "react";
+import { get, ref } from "firebase/database";
+import { firebase_db } from "../../firebase.config";
+
+export default function ParticipantSelector({ participants, setParticipants }) {
+    const [participantDisplayNames, setParticipantDisplayNames] = useState({});
+    const [allChecked, setAllChecked] = useState(false);
+    const db = firebase_db;
+
+    useEffect(() => {
+        const fetchParticipantNames = async () => {
+            const participantNames = {};
+            const promises = Object.keys(participants).map(async (participantKey) => {
+                try {
+                    const userSnapshot = await get(ref(db, `users/${participantKey}`));
+                    if (!userSnapshot.exists()) {
+                        participantNames[participantKey] = "Deleted Account";
+                    } else {
+                        const userData = userSnapshot.val();
+                        participantNames[
+                            participantKey
+                        ] = `${userData.firstName} ${userData.lastName}`;
+                    }
+                } catch (error) {
+                    console.log("uh oh: ", error.message);
+                }
+            });
+            await Promise.all(promises);
+            setParticipantDisplayNames(participantNames);
+        };
+
+        // Call the async function
+        fetchParticipantNames();
+    }, [participants]);
+
+    return (
```

Comment:
Should have checks to uncheck allchecked if you then uncheck one of the boxes, and maybe to automatically check allchecked if all are manually checked

Reply:
So I've modelled it off of how gmail works it, it doesn't uncheck allchecked if you check one of the boxes. Will add in the auto check allchecked though